src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -129,7 +130,6 @@ def can_finish_async(self) -> bool:
         which is harder for users to find. Otherwise, most callbacks should return `True` to avoid
         slowing down Pants from finishing the run.
         """
-        return False
```

Comment:
IMO, default this to `True`, and then don't require folks to set it. I don't think that it matters at a fundamental level, but blocking the run for metrics purposes should be the exception rather than the rule.

Reply:
My concern is that plugin authors will be confused why their logs aren't "showing up" - I had no idea to look at `pants.d/pants.log` until you told me. Personally, I don't think I'm comfortable with defaulting to `True`.

I'm fine with defaulting to `False`, but then it's a missed opportunity perf wise

Reply:
I don't feel strongly about it. Fine either way.